src/components/map/map-legend.tsx

```diff
@@ -45,22 +46,23 @@ export class LegendComponent extends BaseComponent<IProps, IState> {
     const { toggledToSecondary } = this.state;
     const currentLegendType = legendType === "Tephra" && toggledToSecondary ? "Risk" :
                               legendType === "Risk" && toggledToSecondary ? "Tephra" :
-                              legendType;
+                              legendType === "Strain" && toggledToSecondary ? "GPS" :
+                              legendType === "Tephra" ? "Tephra" : "Strain";
     const legend = currentLegendType === "Tephra" ? <TephraLegendComponent onClick={onClick} /> :
                     currentLegendType === "Risk" ? <RiskLegendComponent onClick={onClick} /> :
-                    <StrainLegendComponent onClick={onClick} colorMethod={colorMethod} />;
-    const showToggle = legendType === "Tephra" || legendType === "Risk";
+                    currentLegendType === "Strain" ?
+                    <StrainLegendComponent onClick={onClick} colorMethod={colorMethod} /> :
+                    <GPSLegendComponent onClick={onClick} />;
     return (
       <LegendContainer data-test="key-container">
         {
           legend
         }
         {
-          showToggle &&
           <IconButton
             onClick={this.onLegendModeClick}
             disabled={false}
-            label={`Show ${currentLegendType === "Tephra" ? "Risk" : "Tephra"}`}
+            label={`Show ${currentLegendType === "Tephra" ? "Risk" : currentLegendType === "Risk" ? "Tephra" : currentLegendType === "Strain" ? "GPS" : "Strain"}`}
```

Comment:
Here again I think this has gotten a little too long and too hard to parse. It also again doesn't work on the Monte Carlo tab. I'd define the value higher up using clearer statements, perhaps using the same `secondaryPanel` object you define above.